pkg/models/EndpointDetailsReport/Message.go

```diff
@@ -16,120 +15,146 @@ import (
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.05": func() models.ISODocument {
-		return &camt_052_001_05.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_05], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.06": func() models.ISODocument {
-		return &camt_052_001_06.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_06], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.07": func() models.ISODocument {
-		return &camt_052_001_07.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_07], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.08": func() models.ISODocument {
-		return &camt_052_001_08.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_08], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.09": func() models.ISODocument {
-		return &camt_052_001_09.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_09], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.10": func() models.ISODocument {
-		return &camt_052_001_10.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_10], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.11": func() models.ISODocument {
-		return &camt_052_001_11.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_11], Local: "Document"}}
-	},
-	"urn:iso:std:iso:20022:tech:xsd:camt.052.001.12": func() models.ISODocument {
-		return &camt_052_001_12.Document{XMLName: xml.Name{Space: VersionNameSpaceMap[CAMT_052_001_12], Local: "Document"}}
-	},
+	// EndpointDetailsReport-specific fields
+	Pagenation                         models.MessagePagenation              `json:"pagenation"`
+	BussinessQueryMsgId                string                                `json:"bussinessQueryMsgId"`
+	BussinessQueryMsgNameId            string                                `json:"bussinessQueryMsgNameId"`
+	BussinessQueryCreateDatetime       time.Time                             `json:"bussinessQueryCreateDatetime"`
+	ReportId                           models.ReportType                     `json:"reportId"`
+	ReportingSequence                  models.SequenceRange                  `json:"reportingSequence"`
+	ReportCreateDateTime               time.Time                             `json:"reportCreateDateTime"`
+	AccountOtherId                     string                                `json:"accountOtherId"`
+	TotalCreditEntries                 models.NumberAndSumOfTransactions     `json:"totalCreditEntries"`
+	TotalDebitEntries                  models.NumberAndSumOfTransactions     `json:"totalDebitEntries"`
+	TotalEntriesPerBankTransactionCode []models.TotalsPerBankTransactionCode `json:"totalEntriesPerBankTransactionCode"`
+	EntryDetails                       []models.Entry                        `json:"entryDetails"`
```

Comment:
All of these fields apply to every version of EndpointDetailsReport? (picking a message at random, but how does this deal with added/removed/optional/required changes across versions)